tests/e2escenarios/e2e-test.go

```diff
@@ -0,0 +1,258 @@
+//go:build linux || darwin || dragonfly || solaris || openbsd || netbsd || freebsd
+// +build linux darwin dragonfly solaris openbsd netbsd freebsd
+
+package e2escenarios
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"path"
+	"path/filepath"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/redhat-developer/odo/tests/helper"
+)
+
+var _ = Describe("E2E Test", func() {
+	var commonVar helper.CommonVar
+	var _ = BeforeEach(func() {
+		commonVar = helper.CommonBeforeEach()
+	})
+	var _ = AfterEach(func() {
+		helper.CommonAfterEach(commonVar)
+	})
+
+	checkIfDevEnvIsUp := func(url, assertString string) {
+		Eventually(func() string {
+			url := fmt.Sprintf("http://%s", url)
+			resp, err := http.Get(url)
+			Expect(err).ToNot(HaveOccurred())
+			defer resp.Body.Close()
+
+			body, _ := io.ReadAll(resp.Body)
+			return string(body)
+		}, 60*time.Second, 10*time.Second).Should(Equal(assertString))
+	}
+
+	Context("starting with empty Directory", func() {
+		componentName := helper.RandString(6)
+		var _ = BeforeEach(func() {
+			helper.Chdir(commonVar.Context)
+			Expect(helper.ListFilesInDir(commonVar.Context)).To(BeEmpty())
+		})
+
+		It("should verify developer workflow from empty Directory", func() {
+			deploymentName := "my-component"
+			serviceName := "my-cs"
+			getDeployArgs := []string{"get", "deployment", "-n", commonVar.Project}
+			getSVCArgs := []string{"get", "svc", "-n", commonVar.Project}
+
+			command := []string{"odo", "init"}
+			output, err := helper.RunInteractive(command, nil, func(ctx helper.InteractiveContext) {
+
+				helper.ExpectString(ctx, "Select language")
+				helper.SendLine(ctx, "javascript")
+
+				helper.ExpectString(ctx, "Select project type")
+				helper.SendLine(ctx, "Node.js\n")
+
+				helper.ExpectString(ctx, "Which starter project do you want to use")
+				helper.SendLine(ctx, "nodejs-starter\n")
+
+				helper.ExpectString(ctx, "Enter component name")
+				helper.SendLine(ctx, componentName)
+
+				helper.ExpectString(ctx, "Your new component '"+componentName+"' is ready in the current directory")
+
+			})
+			Expect(err).To(BeNil())
+			Expect(output).To(ContainSubstring("Your new component '" + componentName + "' is ready in the current directory"))
+			Expect(helper.ListFilesInDir(commonVar.Context)).To(ContainElement("devfile.yaml"))
+			Expect(helper.ListFilesInDir(commonVar.Context)).To(ContainElement("server.js"))
+
+			// "execute odo dev and add changes to application"
+			var devSession helper.DevSession
+			var outContents []byte
+			var ports map[string]string
+
+			devSession, outContents, _, ports, err = helper.StartDevMode()
+			helper.ReplaceString(filepath.Join(commonVar.Context, "server.js"), "from Node.js", "from updated Node.js")
+			Expect(err).ToNot(HaveOccurred())
+			helper.MatchAllInOutput(string(outContents), []string{"Watching for changes in the current directory"})
```

Comment:
Here, you are checking for the first "Watching for changes in the current directory", not the one after the sync of server.js happens.

You should use devSession.Waitsync() instead